Userland/Libraries/LibJS/Runtime/JSONObject.cpp

```diff
@@ -335,7 +336,7 @@ String JSONObject::quote_json_string(String string)
             builder.append("\\\\");
             break;
         default:
-            if (code_point < 0x20) {
+            if (code_point < 0x20 || Utf16View::is_high_surrogate(code_point) || Utf16View::is_low_surrogate(code_point)) {
```

Comment:
There is a FIXME above this loop that one can perhaps be removed? (Maybe there are other UTF-16 issues I don't know).

Reply:
I don't know of which UTF-16 issues this is referencing.
AFAIK: This fix isn't really a UTF-16 issue, it's only an issue with ill-formed UTF-16 strings. For example any of thoses strings throws if you try to `encode` or `decode` them in python (`encode` and `decode` happen when converting strings to bytes and reverse, nothing json related).

> Maybe there are other UTF-16 issues I don't know

@mattco98 git is blaming you, could you pls confirm if there are more issues or if that is all ?

Reply:
Looking at the spec I don't see any other issues apart from the fact that there may be UTF-16 input. So it might be referring to that.

Reply:
I know that a really easy to thing to say after the fact.
But this was a 1 hour fix, 1 line fix. (50 minutes of them was me rebuilding the world and fighting with my low C++ skillset).
I belive if this was really all of it @mattco98 would have implemented it.

Reply:
> @mattco98 git is blaming you (https://github.com/SerenityOS/serenity/commit/39576b22385a2e6b6fc4fbf5e90e6b72157e9ee2), could you pls confirm if there are more issues or if that is all ?

I wrote that code a year and a half ago, so the origins of that comment are unfortunately lost to time.

Reply:
Oh yeah don't get me wrong I don't mean to downplay how hard this was to find and fix.
I just saw the FIXME above referring to UTF-16 and since it doesn't explain what should be handled I thought this might be it.
If you feel that it is referring to other problem then just leave it.

Reply:
> I wrote that code a year and a half ago, so the origins of that comment are unfortunately lost to time.

Well thx still :)

Reply:
> If you feel that it is referring to other problem then just leave it.

When writing the FIXME is in the same order of time than writing the fix, I doubt someone would choose to write a fixme.

Reply:
Let's remove the FIXME, this method is now spec compliant and there aren't any other UTF-16 issues to be fixed. Also please add a test :)

Reply:
Done